File: framework/test2/tree/mirror/Mirror.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.Validators;
import soda.unittest.work.TestWork;

import static soda.unittest.LoggerHelper.logger;

class Solution {
    public TreeNode mirror(TreeNode root) {
        if (root == null) {
            return null;
        }
        mirror(root.left);
        mirror(root.right);
        var temp = root.left;
        root.left = root.right;
        root.right = temp;
        return root;
    }
}

public class Mirror implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "mirror");
        // var work = TestWork.forStruct(Struct.class);
        // work.setValidator((e, r) -> {...});
        work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Mirror().get().run();
    }

}

File: framework/test2/tree/mirror/Mirror.scala
import soda.scala.unittest._

import scala.reflect.runtime.universe.typeOf

import soda.scala.leetcode._

import collection.mutable
import util.control.Breaks.{break, breakable}
object Solution {
  def mirror(root: TreeNode): TreeNode = {
    if (root == null) {
      return null
    }
    mirror(root.left)
    mirror(root.right)
    val temp = root.left
    root.left = root.right
    root.right = temp
    root
  }
}

class Solution {}

class Mirror {
  def get(): TestWork = {
    val work = TestWork.forObject(typeOf[Solution], "mirror")
    // val work = TestWork.forStruct(...)
    // work.setValidator((R, R) => Boolean)
    work.compareSerial = true
    work
  }
}

object Mirror extends App {
  new Mirror().get().run()
}
